src/member-api.js

```diff
@@ -572,31 +596,35 @@ export class MemberApi extends TypedEmitter {
     const deviceInfoByConfigCoreId = keyBy(allDeviceInfo, ({ docId }) => docId)
 
     return Promise.all(
-      [...allRoles.entries()].map(async ([deviceId, role]) => {
-        /** @type {MemberInfo} */
-        const memberInfo = { deviceId, role }
-
-        try {
-          const configCoreId = await this.#coreOwnership.getCoreId(
-            deviceId,
-            'config'
-          )
-
-          const deviceInfo = deviceInfoByConfigCoreId.get(configCoreId)
-
-          memberInfo.name = deviceInfo?.name
-          memberInfo.deviceType = deviceInfo?.deviceType
-          memberInfo.joinedAt = deviceInfo?.createdAt
-          memberInfo.selfHostedServerDetails =
-            deviceInfo?.selfHostedServerDetails
-        } catch (err) {
-          // Attempting to get someone else may throw because sync hasn't occurred or completed
-          // Only throw if attempting to get themself since the relevant information should be available
-          if (deviceId === this.#ownDeviceId) throw err
-        }
-
-        return memberInfo
-      })
+      [...allRoles.entries()]
+        .filter(
+          ([_, { roleId }]) => includeLeft || ACTIVE_ROLE_IDS.includes(roleId)
+        )
+        .map(async ([deviceId, role]) => {
```

Comment:
will the size of `allRoles` ever get to a point where it's worth doing more optimal iteration instead of the 4(?) array allocations happening here? i guess `$member.getMany()` may not be a hot path, but what's here stood out to me as something that can be achieved with a single for loop 😄